docs/contributing/code.md

```diff
@@ -61,6 +61,10 @@ In general, we try to avoid use of external libraries.
 - Components should **not** directly access other components -- for example, to publish to MQTT topics.
 - Components are required to dump their configuration using `ESP_LOGCONFIG` in the `dump_config()` method. This method
   is used **exclusively** to **print values** determined during `setup()` -- nothing more.
+- If your component/platform needs to track time, use `App.get_loop_component_start_time()` instead of `millis()`.
+  Reading the hardware is slow; many components/platforms repeatedly checking the hardware slows down the main loop.
+  This method instead returns the start time of the current iteration of the main application loop, which is sufficient
+  for most time-tracking needs.
```

Comment:
```suggestion
- For time tracking, use `App.get_loop_component_start_time()` rather than `millis()`. Hardware time reads are slow,
  and repeated calls from multiple components degrade loop performance. The cached loop start time is sufficient for
  most timing needs.
  
  !!! warning "Time Caching"
      The time is cached at the start of each loop iteration. For long-running operations, you may need to call
      `millis()` to get fresh values. However, if your code runs long enough to need fresh time readings, consider
      breaking it into smaller operations to avoid blocking the main loop.
```

Reply:
That formatted poorly.. let me try again

Reply:
<img width="746" alt="Screenshot 2025-05-19 at 1 09 20 AM" src="https://github.com/user-attachments/assets/a1aa4b07-9e61-4cd1-a286-e24e519bf38f" />